torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -64,6 +65,32 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         prim::RequiresGradCheck /* (...)  -> (..., bool) */,
+         [](const Node* node) -> Operation {
+           std::vector<c10::optional<bool>> rg_props =
+               fmap(node->tys(attr::types), [](const TypePtr& t) {
+                 return t->cast<TensorType>()->requiresGrad();
+               });
+           return [rg_props](Stack* stack) {
+             auto num_inputs = rg_props.size();
+             TORCH_INTERNAL_ASSERT(
+                 stack->size() >= num_inputs && num_inputs > 0);
+             // Check every input's shape against profiled (expected) shape.
+             for (size_t i = 0; i < num_inputs; i++) {
+               auto& input = peek(stack, i, num_inputs);
+               auto t = input.toTensor();
```

Comment:
`auto t` will make a copy of the `Tensor`.  It seems like you should be able to use `auto const& t` but at least do `auto& t`.

Reply:
ah! a good catch! fixed